privaterelay/views.py

```diff
@@ -275,10 +277,20 @@ def _update_all_data(social_account, extra_data, new_email):
             newly_premium = not had_premium and now_has_premium
             no_longer_premium = had_premium and not now_has_premium
             if newly_premium:
+                incr_if_enabled("user_purchased_premium", 1)
                 profile.date_subscribed = datetime.now(timezone.utc)
                 profile.save()
             if no_longer_premium:
                 incr_if_enabled("user_has_downgraded", 1)
+            now_has_phone = profile.has_phone
+            newly_phone = not had_phone and now_has_phone
+            no_longer_phone = had_phone and not now_has_phone
+            if newly_phone:
+                incr_if_enabled("user_purchased_phone", 1)
```

Comment:
TODO?: Re-set the relay number remaining stats? So when someone purchases phones AGAIN after letting their phone subscription lapse, they immediately get texts & minutes back?

(Is my logic here right?)

Reply:
They do not get their minutes back until 30 days plus `profile.date_phone_subscription_checked`. So, if it is a short pause, it may be a few days to reset, and if it is a longer (31+ days) pause, it will be when `update_phone_remaining_stats.py` runs again. If they had an old balance, they can use their old minutes / texts until then.

`date_subscribed_phone` has nothing to do with remaining minutes / texts. I think that is OK?